OpenTabletDriver.UX/Windows/WindowSingleton.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using Eto.Forms;
+
+namespace OpenTabletDriver.UX.Windows
+{
+    public class WindowSingleton<T> where T : Form, new()
+    {
+        public WindowSingleton()
+        {
+            RefreshInternalForm();
+        }
+
+        private T form;
+        private bool isClosed;
+
+        public void ShowSingleton()
+        {
+            if (isClosed)
+                RefreshInternalForm();
+
+            form.Show();
+            form.Focus();
+        }
+
+        private void RefreshInternalForm()
+        {
+            isClosed = false;
+            form = new T();
+            form.Closed += (_, _) => isClosed = true;
+        }
```

Comment:
We could completely omit the need for the `isClosed` field by nulling out the form when it is closed.
If we do it this way, we won't initialize `form` early (only initialized we ever call this function) and potentially reduces memory usage.

Also, why not just call this `Show()`? The class itself implies its a Singleton so the suffix is kinda pointless IMO
```suggestion
        public void Show()
        {
            if (form == null)
            {
                form = new T();
                form.Closed += (_, _) => form = null;
            }

            form.Show();
            form.Focus();
        }
```

Reply:
Code cleaned up to this + switch case for Form and Dialog